cli/pcluster/config/validators.py

```diff
@@ -1420,3 +1433,22 @@ def validate_backup_options(
         fsx_imported_file_chunk_size or fsx_import_path or fsx_export_path or fsx_auto_import_policy
     ) and fsx_automatic_backup_retention_days:
         errors.append("Backups cannot be created on S3-linked file systems")
+
+
+def validate_storage_type_options(
+    errors, fsx_storage_type, fsx_deployment_type, fsx_per_unit_storage_throughput, fsx_drive_cache_type
+):
+    if fsx_storage_type == "HDD":
+        if fsx_deployment_type != "PERSISTENT_1":
+            errors.append("For HDD filesystems, 'deployment_type' must be 'PERSISTENT_1'")
+        if fsx_per_unit_storage_throughput not in [12, 40]:
+            errors.append("For HDD filesystems, 'per_unit_storage_throughput' can only have values of 12 or 40")
+        if not fsx_drive_cache_type:
+            errors.append("For HDD filesystems, the 'drive_cache_type' must be specified")
+    elif fsx_storage_type != "HDD" and fsx_drive_cache_type:
+        errors.append("'drive_cache_type' features can be used only with HDD filesystems")
+    elif not fsx_drive_cache_type and fsx_drive_cache_type:
```

Comment:
This can never happen, you're testing the opposite conditions:
`not fsx_drive_cache_type and fsx_drive_cache_type` --> `not A and A` is not possible :D

Probbaly you meant: `elif not fsx_storage_type and fsx_drive_cache_type:`

In any case this test is useless because the previous check `fsx_storage_type != "HDD"` already includes the `fsx_storage_type = None`.
You can verify it by differentiating the error messages and adding unit tests to address the different cases.

Reply:
You are right! Done :D